drivers/regulator/regulator_shell.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2022 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/drivers/regulator.h>
+#if CONFIG_REGULATOR_PMIC
+#include <zephyr/drivers/regulator/consumer.h>
+#endif
+
+LOG_MODULE_REGISTER(regulator_shell, CONFIG_REGULATOR_LOG_LEVEL);
+
+static int cmd_reg_en(const struct shell *sh, size_t argc, char **argv)
+{
+	struct onoff_client cli;
+	const struct device *reg_dev;
+	int ret;
+
+	if (argc != 2) {
+		shell_print(sh, "usage: %s [dev name]", argv[0]);
+		return -EINVAL;
+	}
+	reg_dev = device_get_binding(argv[1]);
+	if (reg_dev == NULL) {
+		shell_error(sh, "regulator device %s not found", argv[1]);
+		return -ENODEV;
+	}
+
+	ret = regulator_enable(reg_dev, &cli);
+	if (ret < 0) {
+		shell_error(sh, "failed to enable regulator, error %d", ret);
+		return ret;
+	}
+	shell_print(sh, "enabled regulator");
+	return 0;
+}
+
+
+static int cmd_reg_dis(const struct shell *sh, size_t argc, char **argv)
+{
+	const struct device *reg_dev;
+	int ret;
+
+	if (argc != 2) {
+		shell_print(sh, "usage: %s [dev name]", argv[0]);
+		return -EINVAL;
+	}
+	reg_dev = device_get_binding(argv[1]);
+	if (reg_dev == NULL) {
+		shell_error(sh, "regulator device %s not found", argv[1]);
+		return -ENODEV;
+	}
+
+	ret = regulator_disable(reg_dev);
+	if (ret < 0) {
+		shell_error(sh, "failed  to disable regulator, error %d", ret);
+		return ret;
+	}
+	shell_print(sh, "disabled regulator");
+	return 0;
+}
+
+
+#if CONFIG_REGULATOR_PMIC
+static int cmd_pmic_set_vol(const struct shell *sh, size_t argc, char **argv)
+{
+	int lvol, uvol, ret;
+	const struct device *reg_dev;
+
+	if (argc != 4) {
+		shell_print(sh, "usage: %s "
+			"[pmic dev] [lower limit (mV)] [upper limit (mV)]",
+			argv[0]);
+		return -EINVAL;
+	}
```

Comment:
Use `SHELL_CMD_ARG` for automatic argc checks.

Reply:
I've updated all shell commands in this PR to use `SHELL_CMD_ARG`, thanks for the pointer.